### PR TITLE
bug: Fix windows dev experience

### LINKED ISSUE #898: [bug]: Windows contributor workflow is broken
**Describe the bug**
There are several issues when trying to get the app to run locally or install. The installation bit should be handled with #864 but trying to develop locally is a headache.

**To Reproduce**
On a Windows 10 machine, checkout the app and run `yarn install`. Then try to run any of the normal script commands like `yarn run lint` or `yarn run watch:venia` or `yarn run build`.

**Expected behavior**
All root package scripts work including `lint`, `test`, `build`, `watch:venia`, `watch:all`, etc.

**Please complete the following device information:**
 - Device: PC
 - OS: Windows 10
 - NPM version6.4.1
 - Node Version 10.15.1
 - Yarn Version 1.13.0

**Additional context**
This should address the workflow for contributors of _this_ repo. Developers should still be able to add pwa-studio packages like `venia-concept` or `peregrine` and utilize them within their own workflows.

**Problems and Possible solutions**

| Problem | Output | Fix |
| ----------- | ----------- |---|
| Use of env vars breaks commands, ie. `BABEL_ENV`. | |Use [cross-env](https://github.com/kentcdodds/cross-env) for commands that use env variables. |
|`yarn build`| Assuming fix using cross-env, still has issues: [Image from Gyazo](https://gyazo.com/41c57e80c79f1a9d340ae745699aec49) |RootComponent path resolution. |
| `yarn run prettier:validate` |[Image from Gyazo](https://gyazo.com/ffbc9202c6e76bd0674eb553a1948e4f)|Replace `'` with `"`|
| `yarn run watch:venia`| [Image from Gyazo](https://gyazo.com/35fef052c87c162cd0fbfc8c5bfe7a46)|RootComponent path resolution.|
| `yarn test` | This actually runs the tests but there are issues with path resolution/matching ||

### PR DESCRIPTION
**Pull request by sirugh**
## Description
Windows Dev Ex is bork. This is in progress (rebased over #864) just to track the work done so far on it. There's still lots to do!

## Related Issue

\
Closes #898.

## How Has This Been Tested?
1) `yarn install` should work with warnings about certain un-installable deps.
1) `yarn run build` should work now that `cross-env` is added and the paths have been fixed. (_it does but there is an error related to `magento-compatibility.js`_)
1) `yarn test` should work on Mac and Windows. (*it does not*)
1) `yarn prettier` and `yarn lint` should work. (*it does, but it changes a bunch of things*)
1) `yarn run watch:venia` should work. 

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have linked an issue to this PR.
- [ ] I have indicated the change type and relevant package(s).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All CI checks are green (linting, build/deploy, etc).
- [ ] At least one core contributor has approved this PR.

### CODE REVIEW

File: packages/pwa-buildpack/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "description": "Build/Layout optimization tooling and Peregrine framework adapters for the Magento PWA",
   "main": "dist/index.js",
   "scripts": {
-    "build": "babel src --out-dir dist --root-mode 'root' --source-maps --copy-files",
+    "build": "cross-env babel src --out-dir dist --root-mode 'root' --source-maps --copy-files",
```
Comment (sirugh) [inline packages/pwa-buildpack/package.json]: @zetlen do we want to include `BABEL_ENV=production` here like in the peregrine commands?

File: packages/pwa-buildpack/src/Utilities/__tests__/configureHost.spec.js [hunk 2]
```diff
@@ -170,7 +169,7 @@ test('warns about sudo prompt if cert needs to be created', async () => {
     await configureHost({ subdomain: 'best-boss-i-ever-had' });
     expect(console.warn).not.toHaveBeenCalled();
     simulate.certCreated();
-    execa.shell.mockRejectedValueOnce(new Error('wat'));
+    isElevated.mockResolvedValueOnce(false);
```
Comment (sirugh) [inline packages/pwa-buildpack/src/Utilities/__tests__/configureHost.spec.js]: Getting failures that `mockResolvedValueOnce` is not a function.

**Comment by awilcoxa** [comment]
@sharkySharks can you check in with @zetlen for next steps on this PR?

**Comment by sirugh** [comment]
I'm going to pick this back up and see how much work it'll be to resolve the conflicts. If it's not too much effort I think it'd be worth trying to wrap this up and get this working with a big caveat or something like "You can develop with PWA Studio on Windows but certain features will not function. Bugs/issues related to the windows developer workflow may be prioritized lower than other work."

File: yarn.lock [hunk 3]
```diff
@@ -7021,7 +7034,20 @@ exec-sh@^0.3.2:
   resolved "https://registry.yarnpkg.com/exec-sh/-/exec-sh-0.3.2.tgz#6738de2eb7c8e671d0366aea0b0db8c6f7d7391b"
   integrity sha512-9sLAvzhI5nc8TpuQUh4ahMdCrWT00wPWz7j47/emR5+2qEfoZP5zzUXvx+vdx+H6ohhnsYC31iX04QLYJK8zTg==
 
-execa@^0.6.0:
+execa@^0.10.0:
```
Comment (sirugh) [inline yarn.lock]: I had to merge this by hand but I'm not 100% sure I got it right -- how do I regenerate a `yarn.lock` properly?

**Comment by sirugh** [comment]
@zetlen this PR is currently in a semi-working state. I'm able to run the local development (`yarn run watch:venia`) and serve the assets so I think the basic developer workflow could be good to go. That being said, it's not perfect and there are still some things that are broken like tests, prettier, and the build script throws an error. My gut feeling is that we should try to pare this down to be as little a change as possible to get some specific set of commands working for a windows developer who wants to use the pwa-studio. I think we're close, if not there already.

I'd love to talk about this more but I know you've got your hands full. Maybe after Imagine?

**Comment by zetlen** [comment]
A while back, I worked a little too long on a consistent console logging experience for this. I think some of the work can be rescued, though I overdid it with ANSI stuff. Let me just shake the dust off that code and see if you can use any of it.

**Comment by hcmlopes** [comment]
just wanted to chime in and confirm this is working for me. If there is any kind of testing I can do to help ensure this as a permanent fix please let me know.

**Comment by sirugh** [comment]
Closing this PR. Discussion for this functionality should happen in the issue: #898.

**Pull request closed without merging**